Topics/Development_Process/MongoDB_in_Docker.md

```diff
@@ -0,0 +1,110 @@
+# MongoDB in Docker
+
+A quick-start tutorial for using MongoDB in a Docker container.
+
+## Table of Contents
+### [The Official Image](#using-the-official-image)
+### [Alongside Your Application](#setup-alongside-application)
+### [Additional Information](#additional-information-1)
+
+## Using the Official Image
+
+You can find [MongoDB's official images](https://hub.docker.com/u/mongodb) for both MongoDB community and enterprise on docker hub. 
+
+Using the community edition for this example, pull the image by using this command: 
+
+```console
+docker pull mongodb/mongodb-community-server
+```
+
+Alternatively you can also find MongoDB's and other images through Docker Desktop's search bar. 
+
+If you just want to get a database server up and running, the easiest way is to use Docker Desktop's GUI. Navigate to the 'Images' tab on the left hand tab, and find the 'mongodb-community-server' image that you just pulled. Under actions click the run button and it will open up a popup prompting you to input optional settings. The only absolutely necessary field to input to get up and running is the 'Ports' field. [Read about ports here.](./MongoDB_in_Docker.md#a-quick-summary-about-ports) 
+
+In the port field, you can see that Docker Desktop already fills in the container port and control protocol for you (`27017/tcp`), so all we have to do is input the host port. We can use the same port (`27017`) as common.
+
+If you have multiple MongoDB containers running at once, using `27016` or other numbers around is also possible. 
+
+If you have other images running as well, docker will not allow multiple images to use the same port.
+
+Now that you have gotten your first MongoDB container up and running, lets explore the other optional settings.
+
+- Container name is self explanatory, this makes it easier to find and manage running containers
+- Volumes allows for data to persist even when the container is removed/not running. 
+    - This is often ideal when having to make changes to the image and recomposing, as you wouldn't want to lose your databases' data.
+    - In MongoDB's case, the container path **MUST** be '/data/db'
+        - This corresponds to which folder in the container you want to be externally mounted
+    - The Host path can then be any folder you like. This is where the data will live in your local machine. 
+        - If you want to use this data in an another MongoDB container, you can simply remount the host path to the same container path.
+    - The environment variables for MongoDB are generally the root user and password.

```

Comment:
Tab this part left too like the previous environment one

Reply:
Good spot.